src/core/openthread.cpp

```diff
@@ -929,20 +930,20 @@ bool otIsSingleton(void)
     return mEnabled && sThreadNetif->GetMle().IsSingleton();
 }
 
-ThreadError otActiveScan(uint32_t aScanChannels, uint16_t aScanDuration, otHandleActiveScanResult aCallback)
+ThreadError otActiveScan(uint32_t aScanChannels, uint16_t aScanDuration, otHandleActiveScanResult aCallback,
+                         void *aCallbackContext)
 {
-    return sThreadNetif->GetMac().ActiveScan(aScanChannels, aScanDuration, &HandleActiveScanResult,
-                                             reinterpret_cast<void *>(aCallback));
+    return sThreadNetif->GetMac().ActiveScan(aScanChannels, aScanDuration, &HandleActiveScanResult, aCallback,
```

Comment:
I'm not sure this multi-layer callback should be built into `Mac::ActiveScan`.  Can we keep the `aCallback` and `aCallbackContext` within this API shim?